### PR TITLE
Added bold text to package and color to CVE only.

### PR DESCRIPTION
**Pull request by joshuachp**
Formatted the output like `Package {pkg} is affected by {severity} {issues}.` with the pkg in bold and the severity and issue colored.

### CODE REVIEW

**Comment by ilpianista** [comment]
I like the bold text, but I don't like to have the CVEs listed after the risk score. Mainly because you can have multiple CVEs affecting one package and the risk score isn't related to all of them, but to the riskier.

Attaching the "before-after" screenshot:
diff

**Comment by joshuachp** [comment]
Then can  I color each CVE by the respective severity and add the merged risk at the end, as it was before?

**Comment by ilpianista** [comment]
Better yet, I think we don't need the risk score anymore if you colour each CVE differently.

Still, someone's terminal could [not support colours](https://stebalien.github.io/doc/term/term/trait.Terminal.html#tymethod.supports_color), so we should display the risk score only in that case.

**Comment by joshuachp** [comment]
I think the level is still useful. The severity is not so easy to understand from the color and I think it depends mainly on your terminal color scheme. It's also nice to have a summery of the CVEs severity.

**Comment by ilpianista** [comment]
Fine. Then just go ahead by coloring each cve respectively. Thanks!

**Comment by joshuachp** [comment]
I added a test to see how it should look with different CVEs. You can run it with  `cargo test -- --nocapture`.

File: src/main.rs [hunk 1]
```diff
@@ -440,3 +473,72 @@ fn print_avgs(options: &Options, avgs: &BTreeMap<String, avg::AVG>) {
...
+            ("CVE-3".to_string(), enums::Severity::Medium),
+        ],
+        fixed: Some("1.0.0".to_string()),
+        severity: enums::Severity::Unknown,
+        status: enums::Status::Fixed,
+    };
+
+    let avg2 = avg::AVG {
+        issues: vec![
+            ("CVE-1".to_string(), enums::Severity::High),
```
Comment (ilpianista) [inline src/main.rs]: Please name this CVE-4 and CVE-5 otherwise the output is **very** confusing.

File: src/main.rs [hunk 2]
```diff
@@ -324,14 +336,20 @@ fn test_merge_avgs() {
...
+            ("CVE-1".to_string(), enums::Severity::Unknown),
+            ("CVE-2".to_string(), enums::Severity::Unknown),
+        ],
         fixed: Some("1.0.0".to_string()),
         severity: enums::Severity::Unknown,
         status: enums::Status::Fixed,
     };
 
     let avg2 = avg::AVG {
-        issues: vec!["CVE-4".to_string(), "CVE-10".to_string()],
```
Comment (ilpianista) [inline src/main.rs]: Why is this change needed?

File: src/main.rs [hunk 3]
```diff
@@ -440,3 +473,72 @@ fn print_avgs(options: &Options, avgs: &BTreeMap<String, avg::AVG>) {
...
+
+    let options = Options {
+        format: None,
+        quiet: 0,
+        upgradable_only: false,
+    };
+
+    print_avgs(&options, &merged);
+}
+
```
Comment (ilpianista) [inline src/main.rs]: There's an extra line here (run `rustfmt`).

File: src/main.rs [hunk 4]
```diff
@@ -373,42 +391,51 @@ fn test_merge_avgs() {
...
-                        )
-                        .expect("term::writeln failed"),
+                        Some(ref f) => {
+                            t.fg(term::color::RED).expect("term::color::RED failed");
+                            writeln!(
+                                t,
+                                "{}",
+                                f.replace("%n", pkg.as_str()).replace(
+                                    "%c",
+                                    avg.issues.iter().map(|issue| &issue.0).join(",").as_str(),
```
Comment (ilpianista) [inline src/main.rs]: The comma gets coloured while it shouldn't.

File: src/main.rs [hunk 5]
```diff
@@ -417,21 +444,27 @@ fn print_avgs(options: &Options, avgs: &BTreeMap<String, avg::AVG>) {
...
-                                    .replace("%c", avg.issues.iter().join(",").as_str(),)
-                            ),
+                            Some(ref f) => {
+                                t.fg(avg.severity.to_color()).expect("term::fg failed");
+                                writeln!(
+                                    t,
+                                    "{}",
+                                    f.replace("%n", pkg.as_str()).replace(
+                                        "%c",
+                                        avg.issues.iter().map(|issue| &issue.0).join(",").as_str(),
```
Comment (ilpianista) [inline src/main.rs]: The comma gets coloured while it shouldn't.

File: src/main.rs (hunk 2 above)
Comment (joshuachp) [reply src/main.rs]: I changed issue in `src/avg.rs`
```rust
-    pub issues: Vec<String>,
+    pub issues: Vec<(String, enums::Severity)>,
```
to print the severity color.

Comment (ilpianista) [reply src/main.rs]: Sorry, I meant the different CVE name.

File: src/main.rs [hunk 6]
```diff
@@ -373,42 +391,52 @@ fn test_merge_avgs() {
...
+                                write!(t, " {}", v).expect("term::writeln failed");
+                                t.reset().expect("term::stdout failed");
+                                println!(" from the testing repos!");
                             } else if avg.status == enums::Status::Fixed {
-                                writeln!(t, "{} Update to {}!", msg, v)
-                                    .expect("term::writeln failed")
-                            } else {
-                                writeln!(t, "{}", msg).expect("term::writeln failed")
+                                // Print: Update to {}!
+                                print!(" Update to");
```
Comment (ilpianista) [inline src/main.rs]: I would like this to be a mix of `write!`, `t.reset` and `writeln!` without using `print`.

Comment (joshuachp) [reply src/main.rs]: OK, fatto.

**Comment by ilpianista** [comment]
There's something wrong. The package `squid 4.8-2` has 3 vulnerabilities, in particular [CVE-2019-18679](https://security.archlinux.org/CVE-2019-18679) which has a Medium severity, but still it's coloured RED:

Screenshot-Rn6R2y

Also, when using `--format`, the commas are coloured.

File: src/main.rs [hunk 7]
```diff
@@ -145,23 +145,29 @@ fn main() {
 
 /// Converts a JSON to an `avg::AVG`
 fn to_avg(data: &Value) -> avg::AVG {
+    let severity = data["severity"]
```
Comment (ilpianista) [inline src/main.rs]: Ok, this is the reason behind the issue I was describing before.

Here, we set the CVE severity to the severity of the AVG. Which isn't the same.

Comment (joshuachp) [reply src/main.rs]: As far as I can see we don't have that information from the json. So the best option is to return to the start commit 04aa1f5 or close the pull request.

Comment (joshuachp) [reply src/main.rs]: Or print the AVG and color only that. But this is outside the scope of this pull request.

**Pull request closed without merging**